ethereum/helpers/uris.ts

```diff
@@ -26,10 +26,10 @@ export function getURITemplate(
               'value','pre-reveal'
             )
           )
-        ) from ${attributesTable} join ${lookupsTable} where rig_id=ID group by rig_id;`.replace(
-          /(\r\n|\n|\r|\s\s+)/gm,
-          ""
         )
+        from ${attributesTable} join ${lookupsTable}
+        where rig_id=ID
+        group by rig_id;`.replace(/(\r\n|\n|\r|\s\s+)/gm, "")
```

Comment:
No real change here, just formatting.

Reply:
Hmm, it seems the regex that removes spaces etc is sensitive to some ways of formatting i'm doing here. This is causing a test failure. @sanderpick it seems like we should be free to format this however is easy to read. Maybe we need to improve the regex? Or document what the formatting limitations are here? (pls let me know, it's not clear to me yet).

Reply:
The reason why it breaks a test is that the regexp replaces all occurrences of multiple spaces (`\s\s+`) with `""`, with the new formatting in this PR the regexp ends up removing all spaces between `${lookupsTable}` and `where rig_ID...` since they are split into two lines and have multiple spaces between them, where as before it was just one space. The query right returned right now is actually invalid, `...from table2where rig_ID=...`

IMO the tests aren't really helping much they way they are written now though since we have to copy/paste all query changes to the tests, and the string diff output is hard to read. I think we should:

- replace the regexp used to remove spaces and newlines with:
```js
import { init, __wasm } from "@tableland/sqlparser";

const normalize = async (sql: string) => {
  if (__wasm == null) {
    await init();
  }
  return (await sqlparser.normalize(sql)).statements[0];
};

export async function getURITemplate(
  tablelandHost: string,
  attributesTable: string,
  lookupsTable: string,
  pilotsTable: string,
  displayAttributes: boolean
): Promise<string[]> {
  if (!displayAttributes) {
    const uri =
      tablelandHost +
      "/query?extract=true&unwrap=true&s=" +
      encodeURIComponent(
        await normalize(`select json_object(
          ...
```

- update the tests to:
1) parse the `s` query parameter and check that the sql returned is valid by using our sqlparser
2) check the query for keywords like `pre-reveal` instead of exact string matching

Reply:
Oh great call to use the parser here!

Reply:
Yep, we'd have to change the regex to preserve single spaces across line breaks, and then add the single spaces at the end of each line that needs it.

For the first test:

<img width="425" alt="Screenshot 2023-03-01 at 8 37 56 AM" src="https://user-images.githubusercontent.com/361000/222205042-b1cffd14-5375-49a9-86f2-a496d21c0215.png">

For the second test:

<img width="778" alt="Screenshot 2023-03-01 at 8 41 03 AM" src="https://user-images.githubusercontent.com/361000/222205143-a756a5e8-e1f0-49dd-8ecf-71a05f05ed0a.png">

The second test will still fail though because it looks like the expected case wasn't updated in the tests.

That would be the quick fix. But I agree w/ @datadanne that this is hard to maintain... classic case of needing to maintain complex expected results.

Reply:
> Oh great call to use the parser here!

Yeah it makes it much simpler and nicer :) Do you want to integrate that in this PR or should we file an issue and do that work later?

Reply:
oops, type above... the regex is just `/( {2,}|\n)/g` (one space: match two or more, single newline: match)

Reply:
Ok I went ahead with @datadanne's suggestion and updated the URI template generation and tests to use the wasm parser to normalize. Feels good to be able to use whatever formatting of sql that makes reading the easiest!